apps/admin-gui/src/app/vos/pages/member-detail-page/member-overview/member-overview-membership/member-overview-membership.component.ts

```diff
@@ -39,23 +39,51 @@ export class MemberOverviewMembershipComponent implements OnChanges {
   loading: boolean;
   displayedColumns = ['attName', 'attValue'];
 
+  memberStatusChanged = false;
+  statusNotChangedToValid = false;
+
   ngOnChanges(): void {
     this.voMembershipDataSource = new MatTableDataSource<string>(['Status', 'Expiration']);
     this.refreshVoExpiration();
   }
 
+  changeStatus() {
+    const config = getDefaultDialogConfig();
+    config.width = '600px';
+    config.data = {member: this.member, voId: this.vo.id};
+    const oldStatus = this.member.status;
+
+    const dialogRef = this.dialog.open(ChangeMemberStatusDialogComponent, config);
+    dialogRef.afterClosed().subscribe( member => {
+      if (member) {
+        this.member = member;
+        if ((oldStatus === 'VALID' && member.status === 'EXPIRED') || (oldStatus === 'VALID' && member.status === 'DISABLED') || member.status === 'VALID') {
+          this.memberStatusChanged = true;
+          this.statusNotChangedToValid = member.status !== 'VALID';
+          this.changeVoExpiration();
```

Comment:
Do we always want to change the expiration? I think that when you set the status to 'INVALID' or 'DISABLED' this should not appear.